src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/LSPDocumentExtensions.cs

```diff
@@ -26,5 +26,25 @@ public static bool TryGetVirtualCSharpDocument(this LSPDocument hostDocument, ou
             virtualDocument = null;
             return false;
         }
+
+        public static bool TryGetVirtualHtmlDocument(this LSPDocument hostDocument, out VirtualDocument virtualDocument)
```

Comment:
>TryGetVirtualHtmlDocument [](start = 27, length = 25)

Would it make sense to make a generic TryGetVirtualDocument<VirtualDocumentType(...) where VirtualDocumentType : VirtualDocument?

Reply:
Hmm, ya that'd probably be nice

Reply:
Going to do this as part of the initial PR where LSPDocument is defined.